server/src/main/java/com/linecorp/centraldogma/server/internal/thrift/CentralDogmaServiceImpl.java

```diff
@@ -121,22 +136,34 @@ public void listRemovedProjects(AsyncMethodCallback resultHandler) {
     }
 
     @Override
-    public void createRepository(
-            String projectName, String repositoryName, AsyncMethodCallback resultHandler) {
-        handle(executor.execute(Command.createRepository(SYSTEM, projectName, repositoryName)),
+    public void createRepository(String projectName, String repositoryName,
+                                 AsyncMethodCallback resultHandler) {
+        handle(mds.addRepo(SYSTEM, projectName, repositoryName)
+                  .thenCompose(unused -> executor.execute(
```

Comment:
We call `mds` before `CommandExecutor` for `createRepository` and `removeProject` but not for others (`unremoveProject`, `unremoveRepository` and `removeRepository`). Do we need some sort of consitency in call order?

Reply:
I made their call order the same as the call order in HTTP v1 API services. When I applied metadata to HTTP v1 API services first, I thought the call order had to be different for each case. But now I'm thinking that calling `CommandExecutor` before updating metadata is reasonable for all cases. Let me fix them.

Reply:
For `removeProject`, we should update metadata first, because we cannot update metadata if the project is removed.